core/src/main/java/org/kiwix/kiwixmobile/core/main/CoreMainActivity.kt

```diff
@@ -96,6 +98,7 @@ abstract class CoreMainActivity : BaseActivity(), WebViewProvider {
         exitProcess(KIWIX_INTERNAL_ERROR)
       }
     }
+    objectBoxToRoomMigrator.migrateObjectBoxDataToRoom()
```

Comment:
As @kelson42 said we should migrate once. This runs everytime.

Reply:
@gouri-panda, 
* We are migrating once because after migrating we are removing the data from `objectbox` so it won't migrate again it will check if there is any data available for migration then it will run migration otherwise it will skip the process.
* I have not placed the migration available condition here because it needs to create extra variables in `CoreMainActivity`  and needs to place `if` `else` conditions here(for recent searches and we will migrate other stuff as well in the future like NotesEntity etc) so to avoid placing extra conditions we are directly checking in `ObjectBoxToRoomMigrator` class.

Reply:
@MohitMaliFtechiz  Ok. It's not the best approach but we can work around this. We can put a `sharedpPreferce` boolean value that checks whether it has successfully migrated previously or not. We should do this here.
>to avoid placing extra conditions we are directly checking in ObjectBoxToRoomMigrator class.

We should do all the migarations in the `ObjecBoxToMigrator` class.

Reply:
> @MohitMaliFtechiz Ok. It's not the best approach but we can work around this. We can put a sharedpPreferce boolean value that checks whether it has successfully migrated previously or not. We should do this here.

Okay

> We should do all the migarations in the ObjecBoxToMigrator class.

Yes, we are only migrating the data in `ObjectBoxToRoomMigrator` class, migrating condition is handled by the `migrateRecentSearch` method itself.

Reply:
@gouri-panda, I have did the changes you requested.